redaxo/src/core/lib/view.php

```diff
@@ -221,6 +221,7 @@ private static function message($message, $cssClass)
             $cssClassMessage .= ' ' . $cssClass;
         }
 
+        $message = rex_string::sanitizeHtml($message);
```

Comment:
Spontan bin ich mir nicht sicher, ob ich das hier so global tun würde.
Eigentlich hätte ich gesagt, ist hier schon beliebiges HTML erlaubt. Mit onclick-Handler, und so weiter.

Muss ich nochmal drüber nachdenken.

Interessant fände ich auch den Performance-Vergleich.

Reply:
Ergänzung zu Performance: Wir zeigen eigentlich selten viele Messages gleichzeitig an. Daher wahrscheinlich doch nicht so relevant.
(Wobei in yform beim import es teilweise glaube ich vorkommt, dass sehr viele Messages untereinander erscheinen können.)

Reply:
es wahr relativ komfortabel es hier einzubauen. die alternative wäre sehr viele aufrufende pfade in diese methoden rein zu validieren/verifizieren

Reply:
Zu Performance: Ich habe jetzt mal mit der einen Module-Lösch-Meldung getestet, die du oben auch im Screenshot hast. Da kommt sogar schon +24.5% raus: https://blackfire.io/profiles/compare/253b8fc3-82ae-4207-8c20-748bbdf53f41/graph

Reply:
d.h. wie denkst du kann eine alternative aussehen?

Reply:
Muss ich mir genauer anschauen. Aber wenn ich die Fehler so überfliege, habe ich das Gefühl, dass ein wesentliches Problem zum Beispiel ist, dass bei `rex_request('my-var', 'int')` etc. bei der Taint-Analyse scheinbar unser Psalm-Plugin nicht richtig greift.
Also da da ein int rauskommt, sollte das ja safe sein.

Reply:
> dass ein wesentliches Problem zum Beispiel ist, dass bei rex_request('my-var', 'int') etc. bei der Taint-Analyse scheinbar unser Psalm-Plugin nicht richtig greift.
Also da da ein int rauskommt, sollte das ja safe sein.

Ohja, ist mir auch aufgefallen

Reply:
Hmm, wobei ich mir doch nicht sicher bin, jetzt gerade sehe ich solche Messages gar nicht mehr.
Aber vielleicht übersehe ich sie auch in der Masse.

Reply:
du musst im anderen PR schauen:

https://github.com/redaxo/redaxo/pull/4062

dort gibts einige https://github.com/redaxo/redaxo/pull/4062/checks?check_run_id=1426311345

![grafik](https://user-images.githubusercontent.com/120441/99714701-352cd380-2aa6-11eb-80b7-c0e7c80485f3.png)

Reply:
> Zu Performance: Ich habe jetzt mal mit der einen Module-Lösch-Meldung getestet, die du oben auch im Screenshot hast. Da kommt sogar schon +24.5% raus: [blackfire.io/profiles/compare/253b8fc3-82ae-4207-8c20-748bbdf53f41/graph](https://blackfire.io/profiles/compare/253b8fc3-82ae-4207-8c20-748bbdf53f41/graph)

ich hab nen upstream PR geöffnet, der den perf impact unmessbar macht: https://github.com/voku/anti-xss/pull/65

mal sehen ob die Änderung akzeptabel ist.

Reply:
Wobei für mich unabhängig davon noch nicht 100% klar ist, ob ich die Änderung hier so richtig fände.

Reply:
Ich hab mich auch nicht festgelegt.

Perf. Verbesserungen für antixss können uns auch an anderen stellen helfen

Reply:
Jupp, klar. Als ich vorhin das Performance-Ergebnis hatte, dachte ich auch direkt, da muss Markus wohl mal ran ;)

Reply:
> dass bei `rex_request('my-var', 'int')` etc. bei der Taint-Analyse scheinbar unser Psalm-Plugin nicht richtig greift.

Dazu habe ich gestern ein Issue angelegt: https://github.com/vimeo/psalm/issues/4634

Reply:
es gibt ein neues anti-xss release, welches die performance verbessern sollte